src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerHelper.cs

```diff
@@ -0,0 +1,116 @@
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData.Formatter.Serialization
+{
+    /// <summary>
+    /// Helper class for OData Serialization
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1053:StaticHolderTypesShouldNotHaveConstructors")]
+    internal class ODataSerializerHelper
+    {
+        /// <summary>
+        /// Appends instance annotations to an ODataResource.
+        /// </summary>
+        /// <param name="resource">The ODataResource being annotated</param>
+        /// <param name="resourceContext">The context for the resource instance to be annotated.</param>
+        /// <param name="value">The annotations to write.</param>
+        /// <param name="SerializerProvider">The SerializerProvider to use to write annotations</param>
+        internal static void AppendInstanceAnnotations(ODataResourceBase resource, ResourceContext resourceContext, object value, ODataSerializerProvider SerializerProvider)
+        {
+            IODataInstanceAnnotationContainer instanceAnnotationContainer = value as IODataInstanceAnnotationContainer;
+
+            if (instanceAnnotationContainer != null)
+            {
+                IDictionary<string, object> clrAnnotations = instanceAnnotationContainer.GetResourceAnnotations();
+
+                if (clrAnnotations != null)
+                {
+                    foreach (KeyValuePair<string, object> annotation in clrAnnotations)
+                    {
+                        AddODataAnnotations(resource.InstanceAnnotations, resourceContext, annotation, SerializerProvider);
+                    }
+                }
+
+                if (resource.Properties != null)
+                {
+                    foreach (ODataProperty property in resource.Properties)
+                    {
+                        string propertyName = property.Name;
+
+                        if (property.InstanceAnnotations == null)
+                        {
+                            property.InstanceAnnotations = new List<ODataInstanceAnnotation>();
+                        }
+
+                        IDictionary<string, object> propertyAnnotations = instanceAnnotationContainer.GetPropertyAnnotations(propertyName);
+
+                        if (propertyAnnotations != null)
+                        {
+                            foreach (KeyValuePair<string, object> annotation in propertyAnnotations)
+                            {
+                                AddODataAnnotations(property.InstanceAnnotations, resourceContext, annotation, SerializerProvider);
+                            }
+                        }
```

Comment:
```suggestion
                        IDictionary<string, object> propertyAnnotations = instanceAnnotationContainer.GetPropertyAnnotations(propertyName);

                        if (propertyAnnotations != null)
                        {
                            foreach (KeyValuePair<string, object> annotation in propertyAnnotations)
                            {
                                if (property.InstanceAnnotations == null)
                                {
                                    property.InstanceAnnotations = new List<ODataInstanceAnnotation>();
                                }

                                AddODataAnnotations(property.InstanceAnnotations, resourceContext, annotation, SerializerProvider);
                            }
                        }
```
This way we don't initialize the `property.InstanceAnnotations` property unless there are instance annotations to add...